feature-libs/checkout/base/core/facade/checkout-payment.service.ts

```diff
@@ -34,14 +33,22 @@ export class CheckoutPaymentService implements CheckoutPaymentFacade {
   /**
    * Returns the reload triggers for the cardTypes query
    */
-  protected getCardTypesReloadTriggers(): QueryNotifier[] {
-    return [LanguageSetEvent, CurrencySetEvent];
+  protected getCardTypesQueryReloadTriggers(): QueryNotifier[] {
+    return [CheckoutReloadPaymentCardTypesEvent];
+  }
+
+  /**
+   * Returns the reset triggers for the cardTypes query
+   */
+  protected getCardTypesQueryResetTriggers(): QueryNotifier[] {
+    return [CheckoutResetPaymentCardTypesEvent];
   }
 
   protected cardTypesQuery: Query<CardType[]> = this.queryService.create<
     CardType[]
   >(() => this.checkoutPaymentConnector.getCardTypes(), {
-    reloadOn: this.getCardTypesReloadTriggers(),
+    reloadOn: this.getCardTypesQueryReloadTriggers(),
+    resetOn: this.getCardTypesQueryResetTriggers(),
```

Comment:
attached the reset on to have a 'default' event to invalidate the card types query (if ever needed)

Reply:
yep, as long as we don't dispatch it from anywhere (which we don't, I checked ;) )